ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CJavaDriverBasePlugin.java

```diff
@@ -7,111 +7,136 @@
-
-
-    protected final CassJavaDriverManager cassJavaDriverManager;
-    protected Cluster cluster;
-    protected Session session;
-    protected PreparedStatement readPstmt;
-    protected PreparedStatement writePstmt;
-    Long MaxColCount;
-
-    @Inject
-    public CJavaDriverBasePlugin(PropertyFactory propertyFactory, CassJavaDriverManager javaDriverManager) {
-        this.propertyFactory = propertyFactory;
+    protected volatile DataGenerator dataGenerator;
+    protected volatile String ClusterName;
+    protected volatile String KeyspaceName;
+    protected volatile String TableName;
+    protected volatile String ClusterContactPoint;
+    protected volatile String username;
+    protected volatile String password;
+    protected volatile int connections;
+    protected volatile int port;
+    protected volatile ConsistencyLevel WriteConsistencyLevel = ConsistencyLevel.LOCAL_ONE;
+    protected volatile ConsistencyLevel ReadConsistencyLevel = ConsistencyLevel.LOCAL_ONE;
+    protected volatile Long MaxColCount;
+
+    protected volatile Cluster cluster;
+    protected volatile Session session;
+    protected volatile PreparedStatement readPstmt;
+    protected volatile PreparedStatement writePstmt;
+
+
+    /**
+     * Creates an instance of the abstract CJavaDriverBasePlugin class. Subclasses calling this method should use
+     * Guice injection to provide the arguments.
+     * @param javaDriverManager
+     * @param config
+     */
+    protected CJavaDriverBasePlugin(CassJavaDriverManager javaDriverManager, C config) {
         this.cassJavaDriverManager = javaDriverManager;
+        this.config = config;
     }
 
     @Override
     public void init(DataGenerator dataGenerator) throws Exception {
         this.dataGenerator = dataGenerator;
-
-        ClusterName = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.cluster").asString("localhost").get();
-        ClusterContactPoint = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.host").asString("127.0.0.1").get();
-        KeyspaceName = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.keyspace").asString("dev1").get();
-        TableName =propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.cfname").asString("emp").get();
-        port = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE + "cass.host.port").asInteger(9042).get();
-        connections = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.connections").asInteger(2).get();
-
-        username = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.username").asString(null).get();
-        password = propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.password").asString(null).get();
-
-        ReadConsistencyLevel = ConsistencyLevel.valueOf(propertyFactory.getProperty(NdBenchConstants.PROP_NAMESPACE +"cass.readConsistencyLevel").asString(ConsistencyLevel.LOCAL_ONE.toString()).get());
```

Comment:
Did you intentionally leave these out?

Reply:
will put back in

Reply:
same answer is above, different Enum classes and different default values are used for Read/Write ConsistentcyLevel and MaxColCount. I will add a comment here and in implementing classes to indicate what is different.